museek/util/time_analysis.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from datetime import datetime, timedelta
+import ephem
+from museek.time_ordered_data import TimeOrderedData
+
+class TimeAnalysis:
+    """ Class to do time-related analysis. """
+
+    def __init__(self, data: TimeOrderedData):
+        """
+        Initialise
+        :param data: the analysed `TimeOrderedData` object
+        """
+        self.obs_start = datetime.utcfromtimestamp(float(data.original_timestamps[0])) 
+        self.obs_end = datetime.utcfromtimestamp(float(data.original_timestamps[-1])) 
+        self.lat = data.antennas[0].ref_observer.lat
+        self.long = data.antennas[0].ref_observer.long
+
+    def time_difference_to_sunset_sunrise(self, timezone):
+
+        """
+        Calculate the closeness between start/end time and sunset/sunrise time
+        :param timezone: float, the time zone where the observer is located
```

Comment:
I don't understand how a time zone can be represented by a float. Also, I would have thought that `self.lat` and `self.lon` already identify the timezone?

Reply:
It is the offset to utc (in hours), in python it is kind of complicated to calculate the time zone from lat and lon (need install new packages), so I keep it as an input parameter.